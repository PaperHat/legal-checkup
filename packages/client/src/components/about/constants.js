export const HEADING_TEXT = `About`

export const INTRO = `The Legal Check-up is an initiative from Wellington & Hutt Valley Community Law Centre in conjunction with students from PaperHat’s Cowboy team.`

export const INTRO_2 = `The idea of a legal health check has been used all over the world and is based on international research that found that people often have multiple, unaddressed legal issues, but don’t think of them as legal problems. Because of this, people don’t seek legal advice when legal advice could help them.`

export const INTRO_3 = `By asking simple questions about common issues that a lawyer could help, this Checkup helps people identify potential legal issues. As this check up is created by, and for New Zealanders, the questions are focused on the legal issues that New Zealanders are most likely to experience, where legal advice could help sort.`

export const INTRO_4 = `It is a helpful tool to identify legal issues, but does not form legal advice, nor is it guaranteed to identify all issues that a lawyer could help with. Please reach out to your local community law centre if you have any concerns about any legal issues you a facing. You can find information about Community Law services and clinic hours below.`

export const INTRO_5 = `The Legal Checkup does not create a legal obligation between the user, and Community Law, or guarantee legal assistance for the potential legal needs. 
`

export const HEADING_TEXT2 = `Acknowledgements:`

export const INTRO_6 = `
Special acknowledgement to the Halton Legal Clinic in Canada, and the Queensland Public Interest Law Clearing House Inc (QPILCH) along with the National Association of Community Legal Centres (NACLC) who have inspired this initiative.
Interface design by: Hamish Dryer

`
export const INTRO_7 = `Built by the Cowboy hat team from Paperhat`
