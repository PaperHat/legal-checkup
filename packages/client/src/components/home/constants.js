export const HEADING_TEXT = `How's your legal health?`
export const INTRO = `Many people do not think of their everyday problems as being "legal problems" and do not know that they can get help from a lawyer.`
export const INTRO_2 = `This Checkup is to help you identify these problems for yourself.`
