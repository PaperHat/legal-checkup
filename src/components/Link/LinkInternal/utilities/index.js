import { includes } from 'ramda'

import {
  DESKTOP,
  MOBILE,
  TABLET,
  WHITE,
  FOREGROUND_LINK_ACTIVE,
  BORDER_LINK_ACTIVE,
  OFF_WHITE,
  TRANSPARENT
} from '../../../../constants'

export function getLinkColor ({ format = '', active } = {}) {
  // console.log('format', format, 'active', active)
  return includes('Header', format) && active
    ? `${FOREGROUND_LINK_ACTIVE}`
    : `${WHITE}`
}

export function getLinkBorder ({ format = '', active } = {}) {
  if (format === `${DESKTOP}Header` || format === `${TABLET}Header`) {
    return active
      ? `2px solid ${BORDER_LINK_ACTIVE}`
      : `2px solid ${TRANSPARENT}`
  }
  return null
}

export function getLinkBackgroundColor ({ format = '', active } = {}) {
  return format === `${MOBILE}Header` && active ? `${OFF_WHITE}` : null
}

export function getLinkFontWeight ({ format = '', active } = {}) {
  if (format === `${MOBILE}Header` && active) {
    return 900
  }
  return 500
}

export function getLinkFontFamily ({ format = '', active } = {}) {
  if (format === `${MOBILE}Header`) {
    return active === true ? `'Museo Sans Cyrl 900'` : `'Museo Sans Cyrl 500'`
  }
  return `'Museo Sans Cyrl 500'`
}
