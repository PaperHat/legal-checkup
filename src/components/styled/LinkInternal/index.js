import styled from 'styled-components'

const LinkInternal = styled.a`
  cursor: ${props => (props.active ? 'default' : 'pointer')};
`
LinkInternal.displayName = 'StyledLinkInternal'

export default LinkInternal
