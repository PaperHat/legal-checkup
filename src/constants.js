// Formats
export const DESKTOP = 'desktop'
export const MOBILE = 'mobile'
export const TABLET = 'tablet'

export const ABOUT_PAGE = 'aboutPage'
export const TERMS_PAGE = 'termsPage'

// Text
export const FOR_EXAMPLE = 'For example - '
export const WEBSITE = 'Legal Checkup'

// Identifiers for testing
export const accessKeyA = 'accessKeyA'
export const accessKeyB = 'accessKeyB'
export const accessKeyBar = 'accessKeyBar'
export const accessKeyEsc = 'accessKeyEsc'
export const btnYes = 'btnYes'
export const btnNo = 'btnNo'
export const footerWrapper = 'footerWrapper'
export const communityLawUrl = 'communityLawUrl'
export const termsOfUsePath = 'termsOfUsePath'
export const headerHome = 'headerHome'
export const headerCheckup = 'headerCheckup'
export const headerAbout = 'headerAbout'
export const headerHelp = 'headerHelp'
export const headerWrapper = 'wrapper'
export const headerLinkedLogo = 'linkedLogo'
export const headerTitle = 'title'
export const headerNav = 'headerNav'
export const switchId = 'switchId'
export const switchboardHomePath = 'switchboardHomePath'
export const switchboardAboutPath = 'switchboardAboutPath'
export const switchboardCheckupPath = 'switchboardCheckupPath'
export const switchboardHelpPath = 'switchboardHelpPath'
export const switchboardResultsPath = 'switchboardResultsPath'
export const switchboardTermsOfUsePath = 'switchboardTermsOfUsePath'
export const btnNotSure = 'btnNotSure'
export const progressBarId = 'progressBarId'
export const currentTopicId = 'currentTopicId'
export const counter = 'counter'
export const box = 'box'
export const exampleText = 'exampleText'
export const internalLink = 'internalLink'
export const externalLink = 'externalLink'
export const placeholder = 'placeholder'
export const responsive = 'responsive'
export const expanderText = 'expanderText'
export const allGoodText = 'allGoodText'
export const needHelpParagraph1 = 'needHelpParagraph1'
export const needHelpParagraph2 = 'needHelpParagraph2'
export const needHelpExpander = 'needHelpExpander'
export const redoBtn = 'redoBtn'

// colors (hex) - these can be found in the style-guide on Zeplin
// base colors
export const WHITE = `#fff`
export const BLACK = `#1d1b1b`
export const OFF_WHITE = `#eeedea`
export const GREYISH_BROWN = `#444141`
export const BACKGROUND_NAV = GREYISH_BROWN
export const HEADER_BORDER_BOTTOM = GREYISH_BROWN
export const TEXT_COLOR = `#333`

// accent colors
export const BROWNISH_GREY = `#706262`
export const LOGO_BROWNISH = BROWNISH_GREY
export const RESULT_TOPIC_HEADER_FOREGROUND_BUTTON = BROWNISH_GREY
export const RESULT_TOPIC_HEADER_BORDER_TOP = BROWNISH_GREY
export const RESULT_TOPIC_HEADER_BORDER_BOTTOM = BROWNISH_GREY

export const ROUGE = `#b62645`
export const BACKGROUND_BUTTON = ROUGE
export const BUTTON_BORDER_FOCUS = ROUGE
export const FOREGROUND_BUTTON_ACTIVE = ROUGE
export const FOREGROUND_BUTTON_FOCUS_ACTIVE = ROUGE
export const FOREGROUND_LINK_ACTIVE = ROUGE
export const FOREGROUND_LINK_FOCUS_ACTIVE = ROUGE
export const TITLE_TEXT = ROUGE
export const HEADER_TEXT = ROUGE
export const BORDER_LINK_ACTIVE = ROUGE

export const DARK_ROUGE = `#830312`
export const BACKGROUND_BUTTON_FOCUS = DARK_ROUGE
export const BACKGROUND_LINK_FOCUS = DARK_ROUGE
export const DARK_SKY_BLUE = `#4a90e2`
export const FOREGROUND_LINK = DARK_SKY_BLUE
export const TRANSPARENT = `transparent`
export const PINK = `#efcad1`
export const BACKGROUND_BUTTON_ACTIVE = PINK
export const BACKGROUND_BUTTON_FOCUS_ACTIVE = PINK
export const BACKGROUND_LINK_ACTIVE = PINK
export const BACKGROUND_LINK_FOCUS_ACTIVE = PINK
export const RED = `#`

export const PROGRESS_BAR_DIVIDER = 'hsla(0, 0%, 0%, 0.1)'

// new added or like you know just so your mind knows what it's doing
