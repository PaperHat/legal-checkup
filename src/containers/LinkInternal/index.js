import StyledLinkInternal from '@components/styled/LinkInternal'
import { connect } from 'react-redux'
import { push } from 'connected-react-router'

function mapDispatchToProps (dispatch, { to }) {
  return {
    onClick: e => {
      e.preventDefault()

      dispatch(push(to))
    }
  }
}

export default connect(
  undefined,
  mapDispatchToProps
)(StyledLinkInternal)
