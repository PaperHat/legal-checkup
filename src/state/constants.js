// Actions
// These have been 'namespaced' to the Checkup part of the app
// We will eventually have many more actions (e.g., admin actions), so it will be nice
// to keep them all straight
// Note that these are easy to read and understand when you see them in the Redux DevTools
export const CHECKUP_COMPLETE = '@checkup/CHECKUP_COMPLETE'
export const ERROR_QUESTION_INDEX_OUT_OF_BOUNDS =
  '@checkup/ERROR_QUESTION_INDEX_OUT_OF_BOUNDS'
export const NEXT_QUESTION_ACTIVATED = '@checkup/NEXT_QUESTION_ACTIVATED'
export const NEXT_QUESTION_REQUESTED = '@checkup/NEXT_QUESTION_REQUESTED'
export const PREVIOUS_QUESTION_ACTIVATED =
  '@checkup/PREVIOUS_QUESTION_ACTIVATED'
export const PREVIOUS_QUESTION_REQUESTED =
  '@checkup/PREVIOUS_QUESTION_REQUESTED'
export const QUESTION_REQUESTED = '@checkup/QUESTION_REQUESTED'
export const REQUESTED_QUESTION_ACTIVATED =
  '@checkup/REQUESTED_QUESTION_ACTIVATED'
export const USER_RESPONDED_WITH_NO = '@checkup/USER_RESPONDED_WITH_NO'
export const USER_RESPONDED_WITH_NOT_SURE =
  '@checkup/USER_RESPONDED_WITH_NOT_SURE'
export const USER_RESPONDED_WITH_YES = '@checkup/USER_RESPONDED_WITH_YES'
export const QUESTION_SELECTED = '@checkup/QUESTION_SELECTED' // mine
// Response enumeration
export const NO = 'No'
export const NOT_SURE = 'Not sure'
export const YES = 'Yes'

// Page paths
// Always better to use constants for values used in more than one place in the code --
// It ensures that you don't have different values, which causes painful, hard to find bugs
export const HOME_PATH = '/'
export const CHECKUP_PATH = '/checkup'
export const PRIVACY_POLICY_PATH = '/privacy-policy'
export const RESULTS_PATH = '/results'
export const TERMS_OF_USE_PATH = '/terms-of-use'
