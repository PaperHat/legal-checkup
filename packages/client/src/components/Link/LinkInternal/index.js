import styled from 'styled-components'
import styledMap from 'styled-map'

import { DESKTOP, MOBILE, TABLET } from '../../../constants'
import {
  getLinkColor,
  getLinkBorder,
  getLinkBackgroundColor,
  getLinkFontWeight
} from './utilities'

const LinkInternal = styled.span`
  color: ${getLinkColor};
  border-bottom: ${getLinkBorder};
  background-color: ${getLinkBackgroundColor};
  cursor: ${styledMap`
    active: default;
    default: pointer;
  `};
  font-family: 'Museo Sans Cyrl', sans-serif;
  font-weight: ${getLinkFontWeight};

  ${({ format }) =>
    format === `${DESKTOP}Footer` ? 'margin-right: 120px;' : null}

  ${styledMap('format', {
    [`${DESKTOP}Footer`]: `
      font-size: 18px;
      height: 25px;
      margin-left: 140px;
      text-align: center;
      vertical-align: center;
      opacity: 0.5;
      `,
    [`${TABLET}Footer`]: `
      font-size: 15px;
      padding-top: 5px;
      opacity: 0.5;
      `,
    [`${MOBILE}Footer`]: `
      font-size: 15px;
      padding-top: 5px;
      opacity: 0.5;
      `,
    [`${DESKTOP}Header`]: `
      font-size: 20px;
      margin-right: 120px;
    `,
    [`${TABLET}Header`]: `
      font-size: 18px;
      margin-right: 60px;
    `,
    [`${MOBILE}Header`]: `
      line-height: 1;
      font-size: 15px;
      border-radius: 5px 5px 0 0;
      display: inline-block;
      width: -webkit-fill-available;
      text-align: center;
      padding: 12px 0 11px 0;
    `
  })};
`
LinkInternal.displayName = 'StyledLinkInternal'

export default LinkInternal
