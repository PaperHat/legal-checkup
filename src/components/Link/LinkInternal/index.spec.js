import * as React from 'react'
import { create } from 'react-test-renderer'

import { DESKTOP, MOBILE, TABLET } from '../../../constants'

import LinkInternal from './'

describe('components:Link:LinkInternal', () => {
  it(`matches the snapshot when format is ${DESKTOP}Footer`, () => {
    expect(create(<LinkInternal format={`${DESKTOP}Footer`} />).toJSON()).toMatchSnapshot()
  })

  it(`matches the snapshot when format is ${MOBILE}Footer`, () => {
    expect(create(<LinkInternal format={`${MOBILE}Footer`} />).toJSON()).toMatchSnapshot()
  })

  it(`matches the snapshot when format is ${TABLET}Footer`, () => {
    expect(create(<LinkInternal format={`${TABLET}Footer`} />).toJSON()).toMatchSnapshot()
  })

  it(`matches the snapshot when format is ${MOBILE}Header`, () => {
    expect(create(<LinkInternal format={`${MOBILE}Header`} />).toJSON()).toMatchSnapshot()
  })
})
