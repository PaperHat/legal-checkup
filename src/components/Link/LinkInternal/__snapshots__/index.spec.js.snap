// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:Link:LinkInternal matches the snapshot when format is desktopFooter 1`] = `
<span
  className="sc-bdVaJa aROZF"
  format="desktopFooter"
/>
`;

exports[`components:Link:LinkInternal matches the snapshot when format is mobileFooter 1`] = `
<span
  className="sc-bdVaJa cUWsie"
  format="mobileFooter"
/>
`;

exports[`components:Link:LinkInternal matches the snapshot when format is mobileHeader 1`] = `
<span
  className="sc-bdVaJa iSRzMg"
  format="mobileHeader"
/>
`;

exports[`components:Link:LinkInternal matches the snapshot when format is tabletFooter 1`] = `
<span
  className="sc-bdVaJa exlvvr"
  format="tabletFooter"
/>
`;
